abi/cpuid.go

```diff
@@ -0,0 +1,9 @@
+//go:build !amd64 || gccgo
+// +build !amd64 gccgo
+
+package abi
+
+// cpuid only works on amd64
+func cpuid(op uint32) (eax, ebx, ecx, edx uint32) {
```

Comment:
Perhaps instead of defining cpuid this way, define an initCpuid() across this and cpuid_amd64.go that an init() in abi.go can call? It looks like you didn't want to implement init() twice, which is probably a good call.

Naming the arguments and results when you don't need them is a linter violation for us internally, which I'm not sure why it's not complaining in the CI.

Reply:
you are right, `init` is appropriate here